clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -615,13 +639,22 @@ private void producePutOrDeleteToKafka(
     GenericRecord rmdRecord = mergeConflictResult.getRmdRecord();
     final ByteBuffer updatedRmdBytes =
         rmdSerDe.serializeRmdRecord(mergeConflictResult.getValueSchemaId(), mergeConflictResult.getRmdRecord());
-
+    // Since data is not chunked in RT but chunked in VT, creating the key for the small record case or CVM to be
+    // used to persist on disk after producing to Kafka.
+    final byte[] updatedKeyBytes =
+        isChunked ? ChunkingUtils.KEY_WITH_CHUNKING_SUFFIX_SERIALIZER.serializeNonChunkedKey(key) : key;
```

Comment:
This logic seems to be correct, but the implementation is very tricky..
For deletion, we don't use the `updatedKeyBytes` when calling `LeaderProducedRecordContext.newDeleteRecord`, and it works because of this piece of logic in `LeaderProducerCallback`:
https://github.com/linkedin/venice/blob/main/clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/LeaderProducerCallback.java#L148

For `put` record, this function uses `updatedKeyBytes` when calling `newPutRecord` since the logic in `LeaderProducerCallback` won't update key when `chunkedValueManifest` is `null`, that means for large message, it won't update and for regular-size value, it will update it.
This is very confusing..
I guess we should update it in the following way:
1. In `ActiveActiveStoreIngestionTask`, we always use raw `key`.
2. In `LeaderProducerCallback`, we will always update the `keyBytes` field of `LeaderProducedRecordContext`  if `key` is not `null` (which will be updated if chunking is enabled).


What do you think?

Reply:
Yes, this also confused me for a whole day. Today I have the same basic understanding as what you described, but I was not 100% sure though. But I am open to this change, will do.